Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 1631)
+++ CMakeLists.txt	(working copy)
@@ -1,14 +1,28 @@
 project(BayesX)
 cmake_minimum_required(VERSION 2.8.9)
 set(CMAKE_MODULE_PATH ${BayesX_SOURCE_DIR}/share/cmake ${CMAKE_MODULE_PATH})
-set(ANDREA_SRC 
+
+# Add subversion revision to file
+include(FindSubversion)
+if (Subversion_FOUND)
+	Subversion_WC_INFO(${PROJECT_SOURCE_DIR} MY_SVN)
+	set(SVNREVISION "r${MY_SVN_WC_REVISION}")
+else()
+	message( SEND_ERROR "SVN not found." )
+	set(SVNREVISION "UNKNOWN")
+endif()
+configure_file(svnrevision.h.in "${CMAKE_CURRENT_BINARY_DIR}/svnrevision.h")
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+
+set(ANDREA_SRC
 	andrea/baseline.cpp
 	andrea/cox.cpp
 	andrea/IWLS_baseline.cpp
 	andrea/multibaseline.cpp
 	andrea/multistate.cpp
 )
-set(BIB_SRC 
+set(BIB_SRC
 	bib/Random.cpp
 	bib/adminparse_gnu.cpp
 	bib/bandmat.cpp
@@ -54,11 +68,11 @@
 	dag/ia.cpp
 	dag/ia_mixed.cpp
 )
-set(LEYRE_SRC 
+set(LEYRE_SRC
 	leyre/nbinomial.cpp
 	leyre/zip.cpp
 )
-set(MCMC_SRC 
+set(MCMC_SRC
 	mcmc/baseline_reml.cpp
 	mcmc/distribution.cpp
 	mcmc/fullcond.cpp
@@ -90,7 +104,7 @@
 	mcmc/variance_nonp_vector.cpp
 	mcmc/variance_nonp_vector_nigmix.cpp
 )
-set(PSPLINES_SRC 
+set(PSPLINES_SRC
 	psplines/IWLS_pspline.cpp
 	psplines/bsplinemat.cpp
 	psplines/fullcond_pspline_gaussian.cpp
@@ -102,7 +116,7 @@
 	psplines/spline_basis.cpp
 	psplines/spline_basis_surf.cpp
 )
-set(STRUCTADD_SRC 
+set(STRUCTADD_SRC
 	structadd/FC.cpp
 	structadd/FC_hrandom.cpp
 	structadd/FC_hrandom_variance.cpp
@@ -139,7 +153,7 @@
 	structadd/FC_predictive_check.cpp
         structadd/FC_predict_predictor.cpp
 )
-set(SRC 
+set(SRC
 	${ANDREA_SRC}
 	${BIB_SRC}
 	${DAG_SRC}
Index: main.cpp
===================================================================
--- main.cpp	(revision 1631)
+++ main.cpp	(working copy)
@@ -18,9 +18,11 @@
 
 #include "clstring.h"
 #include "adminparse_gnu.h"
+#include "svnrevision.h"
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <string.h>
 #include <dirent.h>
 #include <sys/stat.h>
 #include <unistd.h>
@@ -35,6 +37,14 @@
 
 int main(int argc, char *argv[])
   {
+
+    if(argc>1 && !strcmp(argv[1], "--svn-revision"))
+      {
+        std::cout << SVNREVISION << std::endl;
+        exit(0);
+      }
+
+
   // terminating commands
 //  ST::string* stop1 = new ST::string("quit") ;
 //  ST::string* stop2 = new ST::string("exit") ;
@@ -155,6 +165,11 @@
 
       char *buf;
       buf = readline("BayesX>");
+      // exit if input is EOF
+      if (NULL == buf) {
+        std::cout << std::endl;
+        break;
+      }
       ST::string* s=new ST::string(buf);
       run = a.parse(*s);
 
@@ -178,4 +193,3 @@
 
   return(0);
   }
-
Index: svnrevision.h.in
===================================================================
--- svnrevision.h.in	(revision 0)
+++ svnrevision.h.in	(working copy)
@@ -0,0 +1 @@
+#define SVNREVISION "${SVNREVISION}"
